src/common/state/CMakeLists.txt

```diff
@@ -136,136 +136,140 @@ foreach(obj_name
-    AttributeSubjectMap.C
-    AttributeSubjectSerialize.C
-    Axes2D.C
-    Axes3D.C
-    AxesArray.C
-    AxisAttributes.C
-    AxisLabels.C
-    AxisRestrictionAttributes.C
-    AxisTickMarks.C
-    AxisTitles.C
-    BoxExtents.C
-    CinemaAttributes.C
-    ClientInformation.C
-    ClientInformationList.C
-    ClientMethod.C
-    ColorAttribute.C
-    ColorAttributeList.C
-    ColorControlPoint.C
-    ColorControlPointList.C
-    ColorTableAttributes.C
-    ColorTableManager.C
-    CompactSILRestrictionAttributes.C
-    ConfigManager.C
-    ConstructDataBinningAttributes.C
-    ContourOpAttributes.C
-    DatabaseAttributes.C
-    DatabaseCorrelation.C
-    DatabaseCorrelationList.C
-    DataNode.C
-    DBOptionsAttributes.C
-    DBPluginInfoAttributes.C
-    EngineList.C
-    EngineProperties.C
-    ExportDBAttributes.C
-    Expression.C
-    ExpressionList.C
-    FileOpenOptions.C
-    FontAttributes.C
-    GaussianControlPoint.C
-    GaussianControlPointList.C
-    GlobalAttributes.C
-    GlobalLineoutAttributes.C
-    GlyphTypes.C
-    HostProfileList.C
-    InteractorAttributes.C
-    InternalResampleAttributes.C
-    JSONNode.C
-    KeepAliveRPC.C
-    KeyframeAttributes.C
-    LaunchProfile.C
-    LaunchRPC.C
-    LightAttributes.C
-    LightList.C
-    Line.C
-    LinearTransformAttributes.C
-    LineGlyphAttributes.C
-    MachineProfile.C
-    MapNode.C
-    MaterialAttributes.C
-    MeshManagementAttributes.C
-    MessageAttributes.C
-    MovieAttributes.C
-    NameschemeAttributes.C
-    NamespaceAttributes.C
-    Observer.C
-    ObserverToCallback.C
-    PickAttributes.C
-    PickVarInfo.C
-    PlaneAttributes.C
-    Plot.C
-    PlotInfoAttributes.C
-    PlotList.C
-    PlotQueryInfo.C
-    PluginManagerAttributes.C
-    PointAttributes.C
-    PointGlyphAttributes.C
-    PrinterAttributes.C
-    ProcessAttributes.C
-    QueryAttributes.C
-    QueryList.C
-    QueryOverTimeAttributes.C
-    QuitRPC.C
-    RenderingAttributes.C
-    SaveWindowAttributes.C
-    SaveSubWindowAttributes.C
-    SaveSubWindowsAttributes.C
-    SelectionList.C
-    SelectionProperties.C
-    SelectionSummary.C
-    SelectionVariableSummary.C
-    SimulationUIValues.C
-    SILArrayAttributes.C
-    SILAttributes.C
-    SILMatrixAttributes.C
-    SILRestrictionAttributes.C
-    SimilarityTransformAttributes.C
-    SimpleObserver.C
-    SimulationCommand.C
-    SingleAttributeConfigManager.C
-    SphereAttributes.C
-    StatusAttributes.C
-    Subject.C
-    SurfaceFilterAttributes.C
-    SyncAttributes.C
-    ThresholdOpAttributes.C
-    TimeFormat.C
-    Variant.C
-    View2DAttributes.C
-    View3DAttributes.C
-    ViewAttributes.C
-    ViewAxisArrayAttributes.C
-    ViewCurveAttributes.C
-    ViewerClientAttributes.C
-    ViewerClientInformation.C
-    ViewerClientInformationElement.C
-    VisItRPC.C
-    VisualCueInfo.C
-    VisualCueList.C
-    WindowAttributes.C
-    WindowInformation.C
-    Xfer.C
-    XMLNode.C)
+visit_append_list(
```

Comment:
Just curious...does CMake support a short-hand notation for *all .C files in this directory* instead of having to list them all and then constantly edit that list as we had or delete them.

Reply:
There is not shortcut, except use of file(GLOB) which is highly discouraged.

Reply:
Why is `file(GLOB...)` highly discouraged?

Its really too bad because it seems like the CMake equivalent of GNU Make's [wildcard function and pattern rules](https://chat.openai.com/share/f533e6b6-80b5-4ea2-b8ec-ab1a8801a17b)

Reply:
I think it has to do with CMake's ability to track whether a source file has changed.  I don't like it myself because I like looking at the file and knowing exactly what is being added to the project.